bluesky_darkframes/__init__.py

```diff
@@ -190,6 +190,10 @@ def clear(self):
     def __call__(self, plan):
         "Preprocessor: Takes in a plan and creates a modified plan."
 
+        if self._disabled:
+            logger.info("%r is disabled, will act as a no-op", self)
+            return (yield from plan)
+
         def insert_dark_frame(force_read, msg=None):
             # Acquire a fresh Snapshot if we need one, or retrieve a cached one.
             state = {}
```

Comment:
I'm just reading through this for the first time, what is the intention of the variable `state`?  Now that this PR removes caching of `state`, has the purpose of `state` changed? Is there a better variable name now?

Reply:
Maybe updating the doc strings of the methods that have a `state` argument would be helpful.

Reply:
State is still cached, but we no long cache information about which state was used most recently, because it turns out we don’t need to. I’ll add some code comments to clarify how this operates.